pkg/server/singleprocess/service_task_test.go

```diff
@@ -354,3 +354,85 @@ func TestServiceTask_ListTaskFilters(t *testing.T) {
 
 	})
 }
+
+func TestServiceTask_CancelTask(t *testing.T) {
+	ctx := context.Background()
+
+	// Create our server
+	db := testDB(t)
+	impl, err := New(WithDB(db))
+	require.NoError(t, err)
+	client := server.TestServer(t, impl)
+
+	// Initialize our app
+	TestApp(t, client, serverptypes.TestJobNew(t, &pb.Job{
+		Application: &pb.Ref_Application{
+			Application: "a_test",
+			Project:     "p_test",
+		},
+	}).Application)
+
+	// Create, should get an ID back
+	jobResp, err := client.QueueJob(ctx, &pb.QueueJobRequest{
+		Job: serverptypes.TestJobNew(t, nil),
+	})
+	require.NoError(t, err)
+	require.NotNil(t, jobResp)
+	require.NotEmpty(t, jobResp.JobId)
+	startJobId := jobResp.JobId
+
+	jobResp, err = client.QueueJob(ctx, &pb.QueueJobRequest{
+		Job: serverptypes.TestJobNew(t, nil),
+	})
+	require.NoError(t, err)
+	require.NotNil(t, jobResp)
+	require.NotEmpty(t, jobResp.JobId)
+	runJobId := jobResp.JobId
+
+	jobResp, err = client.QueueJob(ctx, &pb.QueueJobRequest{
+		Job: serverptypes.TestJobNew(t, nil),
+	})
+	require.NoError(t, err)
+	require.NotNil(t, jobResp)
+	require.NotEmpty(t, jobResp.JobId)
+	stopJobId := jobResp.JobId
+
+	resp, err := client.UpsertTask(ctx, &pb.UpsertTaskRequest{
+		Task: serverptypes.TestValidTask(t, &pb.Task{
+			TaskJob:  &pb.Ref_Job{Id: runJobId},
+			StartJob: &pb.Ref_Job{Id: startJobId},
+			StopJob:  &pb.Ref_Job{Id: stopJobId},
+		}),
+	})
+	require.NoError(t, err)
+	taskId := resp.Task.Id
+
+	// Create, should get an ID back
+	t.Run("cancel existing by task id", func(t *testing.T) {
+		require := require.New(t)
+
+		type JobReq = pb.QueueJobRequest
+
+		// Get, should return a task
+		resp, err := client.GetTask(ctx, &pb.GetTaskRequest{
+			Ref: &pb.Ref_Task{
+				Ref: &pb.Ref_Task_Id{
+					Id: resp.Task.Id,
+				},
+			},
+		})
+		require.NoError(err)
+		require.NotNil(resp.Task)
+		require.NotEmpty(resp.Task.Id)
+		require.Equal(taskId, resp.Task.Id)
+
+		_, err = client.CancelTask(ctx, &pb.CancelTaskRequest{
+			Ref: &pb.Ref_Task{
+				Ref: &pb.Ref_Task_Id{
+					Id: resp.Task.Id,
+				},
+			},
+		})
+		require.NoError(err)
+	})
```

Comment:
This test only verifies that we can call CancelTask and not receive an error. It doesn't seem to actually test if the task gets canceled. I think we should also query or list the task and verify it is cancelled.

Reply:
We do this already in the state tests! It felt weird to test that given it was already covered there, but I can!

Reply:
I find it valuable in that we're connecting the dots that calling this service endpoint actually cancels a task, otherwise I feel like we're just testing that the endpoint exists, not that it actually does anything. Not a huge deal if we leave it as is

Reply:
Sounds good - I can add a few more lines of code to test this, should be pretty easy! Working on that now